influxql/query_manager.go

```diff
@@ -162,11 +223,51 @@ func (qm *defaultQueryManager) waitForQuery(qid uint64, timeout time.Duration, c
 
 	select {
 	case <-closing:
+		qm.mu.Lock()
+		query, ok := qm.queries[qid]
+		qm.mu.Unlock()
```

Comment:
Can you move these 3 lines to a helper function? e.g. `defaultQueryManager.Query(qid)`